test/distributed/test_device_mesh.py

```diff
@@ -756,6 +757,19 @@ def test_get_all_submeshes(self):
             all(submesh.mesh.numel() == 2 for submesh in all_submeshes), True
         )
 
+    @with_comms
+    def test_mesh_slice_fake_tensor_mode(self):
+        mesh_shape = (2, self.world_size // 2)
+        mesh_dim_names = ("DP", "TP")
+        mesh_2d = init_device_mesh(
+            self.device_type, mesh_shape, mesh_dim_names=mesh_dim_names
+        )
+
+        with FakeTensorMode():
+            dp_mesh = mesh_2d["DP"]
```

Comment:
hmm, at least we probably don't want to break tracing, so i guess we do want this to work?  

is there some problem with it currently?

Reply:
Right now the mesh slicing op (create_sub_mesh call) cannot work under FakeTenorMode since it includes actual tensor computation besides shape propagation. 

Try running the test in this PR will produce an error.

Reply:
Can we materialize the tensor? This is a tensor that we do know the content.

Reply:
https://github.com/pytorch/pytorch/pull/142287